tests/test_weather.py

```diff
@@ -0,0 +1,76 @@
+from src.weather import WeatherCall
+from pytest_mock import MockerFixture
+
+def test_check_key_passes(mocker: MockerFixture):
+    '''test the check_key function when the key is correct'''
+    #arrange
+    mock_json_return = '200'
+    mocker.patch("src.weather.WeatherCall.check", return_value = mock_json_return )
+    try:
+        #act
+        WeatherCall.check_key()
+    except TypeError:
+        #assert
+        assert False
+
+def test_check_key_fail(mocker: MockerFixture):
+    '''tests the check_key function when the key is incorrect'''
+    #arrange
+    mock_json_return = '401'
+    mocker.patch("src.weather.WeatherCall.check", return_value = mock_json_return )
+    try:
+        #act
+        WeatherCall.check_key()
+    except TypeError:
+        #assert
+        assert True
+
+def test_get_coordinates_passes(mocker: MockerFixture):
+    '''test the get_coordinates function when the response gives the correct json'''
+    #arrange
+    mock_json_return = [{"name":"Rome","local_names":{"nl":"Rome","eu":"Erroma","am":"ሮማ","pt":"Roma","af":"Rome","fy":"Rome","mr":"रोम","cv":"Рим","sg":"Rome","ja":"ローマ","pl":"Rzym","hy":"Հռոմ","kl":"Roma","sq":"Roma","ks":"روم","ia":"Roma","mg":"Roma","id":"Roma","ca":"Roma","ko":"로마","ga":"An Róimh","es":"Roma","th":"โรม","he":"רומא","se":"Roma","ml":"റോം","cu":"Римъ","uz":"Rim","da":"Rom","lb":"Roum","tg":"Рим","bg":"Рим","sr":"Рим","hi":"रोम","my":"ရောမမြို့","mt":"Ruma","mk":"Рим","fr":"Rome","cr":"ᖌᒪ","nn":"Roma","so":"Roma","ie":"Roma","oc":"Roma","gn":"Róma","bi":"Rome","cy":"Rhufain","be":"Рым","sv":"Rom","ne":"रोम","sh":"Rim","mi":"Rōma","os":"Ром","ar":"روما","sa":"रोमा","lv":"Roma","ee":"Rome","tk":"Rim","en":"Rome","ru":"Рим","ty":"Roma","la":"Roma","ka":"რომი","yo":"Rómù","io":"Roma","uk":"Рим","de":"Rom","fi":"Rooma","gd":"An Ròimh","si":"රෝමය","rn":"Roma","vo":"Roma","no":"Roma","sc":"Roma","et":"Rooma","fa":"رم","ku":"Roma","ht":"Ròm","it":"Roma","gv":"Yn Raue","rw":"Roma","ta":"உரோமை நகரம்","tr":"Roma","kk":"Рим","sk":"Rím","te":"రోమ్","qu":"Roma","sw":"Roma","kn":"ರೋಮ","ay":"Roma","su":"Roma","lt":"Roma","bs":"Rim","kv":"Рим","is":"Róm","yi":"רוים","rm":"Roma","an":"Roma","gl":"Roma","hr":"Rim","el":"Ρώμη","ur":"روم","li":"Roeme","sl":"Rim","tl":"Roma","ug":"رىم","sd":"روم","az":"Roma","zh":"羅馬/罗马","fo":"Róm","co":"Roma","bo":"རོ་མ།","mn":"Ром","vi":"Rô-ma","tt":"Рим","bn":"রোম","ba":"Рим","br":"Roma","ln":"Roma","hu":"Róma","jv":"Roma","cs":"Řím","ro":"Roma","na":"Roma","eo":"Romo"},"lat":41.8933203,"lon":12.4829321,"country":"IT","state":"Lazio"}]
+    mocker.patch("src.weather.WeatherCall.geo_response", return_value = mock_json_return )
+    try:
+        #act
```

Comment:
Perchè lo richiami qui e lo richiami sotto? Ti direi di sostiture la riga 35 con la riga con l'assert  in basso.
O Altrimenti salvare la response e fare l'assert su di questa